### PR TITLE
fix: increase the event observer max body size

### PR DESCRIPTION
**Pull request by matteojug**
## Description

Closes: https://github.com/stacks-network/sbtc/issues/1186

## Changes

Increase the body max size to 8MB for the event observer endpoint. Also add some codegen leftover from some other PR.

## Testing Information

Add 2 tests to check the limit boundary.

## Checklist:

- [X] I have performed a self-review of my code
- [X] My changes generate no new warnings
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: signer/src/api/new_block.rs [hunk 1]
```diff
@@ -914,4 +928,53 @@ mod tests {
         assert_eq!(db.rotate_keys_transactions.len(), 1);
         assert!(db.rotate_keys_transactions.get(&txid).is_some());
     }
+
+    #[test_case(EVENT_OBSERVER_BODY_LIMIT, true; "event within limit")]
+    #[test_case(EVENT_OBSERVER_BODY_LIMIT+1, false; "event over limit")]
```
Comment (djordon) [inline signer/src/api/new_block.rs:933]: style nit that rustfmt cannot see:
```suggestion
    #[test_case(EVENT_OBSERVER_BODY_LIMIT + 1, false; "event over limit")]
```

File: signer/src/api/new_block.rs [hunk 2]
```diff
@@ -914,4 +928,53 @@ mod tests {
...
+
+        let state = ApiState { ctx: ctx.clone() };
+        let app = get_router().with_state(state);
+
+        let db = ctx.inner_storage();
+        // We don't have anything here yet
+        assert!(db.lock().await.rotate_keys_transactions.is_empty());
+
+        let mut event: String = " ".repeat(event_size - ROTATE_KEYS_WEBHOOK.len());
+        event.push_str(ROTATE_KEYS_WEBHOOK);
```
Comment (djordon) [inline signer/src/api/new_block.rs:962]: Nice

**Review by djordon — APPROVED** [review]
Looks good, and nice test.
